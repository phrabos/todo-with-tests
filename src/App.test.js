import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import App from './App';

test('renders learn react link', async () => {
	const { container } = render(<App />);
	const input = screen.getByPlaceholderText('new todo');
	expect(input).toBeInTheDocument();

	userEvent.type(input, 'new todo 1');
	expect(input).toHaveValue('new todo 1');

	const button = screen.getByRole('button', { name: 'submit todo' });
	expect(button).toBeInTheDocument();

	userEvent.click(button);

	userEvent.type(input, '{selectall}{del}new todo 2');
	expect(input).toHaveValue('new todo 2');

	userEvent.click(button);

	const li = screen.getAllByText('new todo', { exact: false });
	expect(li.length).toBe(2);

	const toDelete = screen.getByText('new todo 1');
	userEvent.click(toDelete);
	expect(toDelete).not.toBeInTheDocument();

	expect(container).toMatchSnapshot();
});
