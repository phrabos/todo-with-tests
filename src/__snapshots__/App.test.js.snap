// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div>
  <form>
    <input
      aria-label="todo input"
      placeholder="new todo"
      type="text"
      value=""
    />
    <button
      aria-label="submit todo"
      type="submit"
    >
      add todo
    </button>
  </form>
  <ul>
    todo list
    <li>
      new todo 2
    </li>
  </ul>
</div>
`;
